tracing/src/test/java/com/palantir/remoting1/tracing/AsyncSlf4jSpanObserverTest.java

```diff
@@ -65,17 +67,51 @@ public void after() {
     }
 
     @Test
-    public void testSanity() throws Exception {
+    public void testJsonFormatToLog() throws Exception {
         DeterministicScheduler executor = new DeterministicScheduler();
         Tracer.subscribe("foo", AsyncSlf4jSpanObserver.of(executor));
         Tracer.startSpan("operation");
-        Tracer.completeSpan();
+        Span span = Tracer.completeSpan().get();
         verify(appender, never()).doAppend(any(ILoggingEvent.class)); // async logger only fires when executor runs
 
         executor.runNextPendingCommand();
         assertThat(executor.isIdle()).isTrue();
         verify(appender).doAppend(event.capture());
-        assertThat(event.getValue().getFormattedMessage()).containsSequence("traceId=" + Tracer.getTraceId());
+        assertThat(event.getValue().getFormattedMessage())
+                .isEqualTo(AsyncSlf4jSpanObserver.ZipkinCompatibleSerializableSpan.fromSpan(span).toJson());
         verifyNoMoreInteractions(appender);
     }
+
+    @Test
+    public void testLogFormatIsZipkinCompatible() throws Exception {
+        Span span = Span.builder()
+                .traceId(Tracers.longToPaddedHex(42L))
+                .parentSpanId(Tracers.longToPaddedHex(123456789L))
+                .spanId(Tracers.longToPaddedHex(234567890L))
+                .operation("op")
+                .startTimeMicroSeconds(43L)
+                .durationNanoSeconds(43001L)
+                .build();
+        zipkin.Span zipkinSpan = zipkin.Span.builder()
+                .traceId(42L)
+                .parentId(123456789L)
+                .id(234567890L)
+                .name("op")
+                .timestamp(43L)  // micro-seconds
+                .duration(44L)  // micro-seconds, rounded up
+                .build();
+        String expectedString = new String(Codec.JSON.writeSpan(zipkinSpan), StandardCharsets.UTF_8);
+        String actualString = AsyncSlf4jSpanObserver.ZipkinCompatibleSerializableSpan.fromSpan(span).toJson();
+        assertThat(actualString).isEqualTo(expectedString);
+    }
+
+    @Test
+    public void testNanoToMicro() throws Exception {
+        assertThat(AsyncSlf4jSpanObserver.ZipkinCompatibleSerializableSpan.nanoToMicro(0)).isEqualTo(0);
```

Comment:
Per https://github.com/openzipkin/zipkin-api/blob/master/zipkin-api.yaml#L281 I think you have to round up to 1 microsecond

See also https://github.com/openzipkin/zipkin/issues/1174